### ISSUE TITLE
Typescript errors in question route

### ISSUE THREAD

**Issue by jrhender** [StampyAI/stampy-ui#632]
When running `lint`, type errors are reported:

```
❯ npm run lint

> lint
> tsc && npm run prettier && npm run eslint

app/routes/questions.$questionId.$.tsx:64:10 - error TS2339: Property 'question' does not exist on type 'JsonifyObject<Pick<DeferredData, "init"> & { data: { question: Promise<Response | { data: Question; timestamp: string; }>; tags: Promise<Response | Tag[]>; }; readonly [typedDeferredDataBrand]: "TypedDeferredData"; }>'.

64   const {question, tags} = useLoaderData<typeof loader>()
            ~~~~~~~~

app/routes/questions.$questionId.$.tsx:64:20 - error TS2339: Property 'tags' does not exist on type 'JsonifyObject<Pick<DeferredData, "init"> & { data: { question: Promise<Response | { data: Question; timestamp: string; }>; tags: Promise<Response | Tag[]>; }; readonly [typedDeferredDataBrand]: "TypedDeferredData"; }>'.

64   const {question, tags} = useLoaderData<typeof loader>()
                      ~~~~

app/routes/questions.$questionId.$.tsx:81:20 - error TS7006: Parameter 'val' implicitly has an 'any' type.

81     question.then((val) => {
                      ~~~


Found 3 errors in the same file, starting at: app/routes/questions.$questionId.$.tsx:64
```

These are observed in CI as well: https://github.com/StampyAI/stampy-ui/actions/runs/8961325998/job/24608811709?pr=623

**Comment by jrhender** [comment]
What's strange to me is that recent `PR check` CI jobs (which run `lint` and the typescript compilation) have passed https://github.com/StampyAI/stampy-ui/actions/runs/8935500699/job/24544106416 , without anything related to this code area changing.

**Comment by jrhender** [comment]
I assume that this is just a typescript issue as I believe this page is working at runtime. Furthermore, `npm run dev` and `npm run build` work, which suggests that `remix` is able to compile the code. Maybe a direction of investigation is what the differences between `remix` and `tsc` are, with regards to typescript compilation.

**Comment by jrhender** [comment]
This seems to be fixed when testing on bf3c1d5551e09b9c5bb153174ec513d2b4591cb7 . I don't understand the root cause but I think that understanding the root cause is not important as it is unlikely to be related to other issues and the impact is small if it reoccurs. Therefore, I'm closing this issue.
